transformations/style_paraphraser/transformation.py

```diff
@@ -0,0 +1,251 @@
+import re
+
+import nltk
+import numpy as np
+import torch
+from transformers import GPT2LMHeadModel, GPT2Tokenizer
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from transformations.style_paraphraser.paraphraser_helpers.style_paraphraser import (
+    GPT2ParentModule,
+    Instance,
+    _beam_search,
+    _sample_sequence,
+)
+
+"""
+    Note: This codebase is based upon, adapted and refactored from code
+    from this repository:
+    https://github.com/martiansideofthemoon/style-transfer-paraphrase
+
+"""
+
+MODELS_SUPPORTED = {
+    "Bible": "filco306/gpt2-bible-paraphraser",
+    "Basic": "filco306/gpt2-base-style-paraphraser",
+    "Shakespeare": "filco306/gpt2-shakespeare-paraphraser",
+    "Tweets": "filco306/gpt2-tweet-paraphraser",
+    "Switchboard": "filco306/gpt2-switchboard-paraphraser",
+    "Romantic poetry": "filco306/gpt2-romantic-poetry-paraphraser",
+}
+
+MAX_PARAPHRASE_LENGTH = 100
+
+BASE_CONFIG = {
+    "max_total_length": MAX_PARAPHRASE_LENGTH,
+    "max_prefix_length": int(MAX_PARAPHRASE_LENGTH / 2),
+    "max_suffix_length": int(MAX_PARAPHRASE_LENGTH / 2),
+}
+
+
+class StyleTransferParaphraser(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+    heavy = True
+    keywords = [
+        "model-based",
+        "transformer-based",
+        "tokenizer-required",
+        "unnatural-sounding",
+        "unnaturally-written",
+        "possible-meaning-alteration",
+        "high-coverage",
+    ]
+    """
+        Style transfer paraphraser, using a GPT2-model of choice.
+
+        Args:
+            style : str
+                The style to use. Options include Bible, Shakespeare, Basic, Romantic Poetry, Switchboard and Tweets.
+            device : device to use for computations.
+                Default: None, and it will then resort to CUDA if available, else CPU.
+            upper_length :
+                The maximum length.
+                Options: "eos" or "same_N" (e.g., "same_5"), where N will be the max_length.
```

Comment:
Does the the "same_N" mean `max_length = length of input + N`, and "eos" mean `max_length = length of input`?

It would be good to add more details to make it clearer.

Reply:
Yes! Fixed!

Reply:
Great point!